CMakeLists.txt

```diff
@@ -165,6 +165,56 @@ else()
   set(UUID_LIBRARIES "uuid")
 endif()
 
+if(NOT USE_SYSTEM_CURL)
```

Comment:
May have to add a conditional here to not build if curl support is completely disabled.

Reply:
Agreed, with the caveat that the boolean here has gotten a little muddy due to various hooks into cURL. I'll see what I can do to make a readable bool along the lines of NEEDS_CURL.